hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogImpl.java

```diff
@@ -326,9 +337,18 @@ public void close() {
   }
 
   private void getTransaction(DeletedBlocksTransaction tx,
-      DatanodeDeletedBlockTransactions transactions,
-      Set<ContainerReplica> replicas,
-      Map<DatanodeID, Map<Long, CmdStatus>> commandStatus) {
+                                       DatanodeDeletedBlockTransactions transactions,
+                                       Set<ContainerReplica> replicas,
+                                       Map<DatanodeID, Map<Long, CmdStatus>> commandStatus) {
+
+    // Check if all replicas satisfy the maxBlocksPerDatanode condition
+    if (!replicas.stream().allMatch(replica -> {
```

Comment:
And if the ((DN count) * maxDeleteBlocksPerDatanode) < blockDeletionLimit, the DeletedBlockLogImpl#getTransactions will iterator all the data in the table.